src/parser/Expression.ts

```diff
@@ -939,6 +939,56 @@ export class NewExpression extends Expression {
     }
 }
 
+/**
+ * This expression changes a call expression into a createObject call,
+ * eg `new roSGNode("Poster")` transpiles to `CreateObject("roSGNode", "Poster")`
+ */
+export class NewCreateObjectExpression extends Expression {
+    constructor(
+        readonly newKeyword: Token,
+        readonly call: CallExpression
+    ) {
+        super();
+        this.range = util.createRangeFromPositions(this.newKeyword.range.start, this.call.range.end);
+    }
+
+    /**
+     * The name of the component to initialize (with optional namespace prefixed)
+     */
+    public get componentName() {
+        //the parser guarantees the callee of a new statement's call object will be
+        //a NamespacedVariableNameExpression
+        return (this.call.callee as NamespacedVariableNameExpression);
+    }
+    public readonly range: Range;
+
+    public transpile(state: BrsTranspileState) {
```

Comment:
This logic should probably live in BrsFilePreTranspileProcessor plugin function. Either that, or we have an if statement inside NewExpression that checks whether this is a ro[Whatever] or a class.

Reply:
I don't understand how that works or why it should be in BrsFilePreTranspileProcessor.... Can you elaborate?

Reply:
I've been slowly moving lots of functionality into the BscPlugin for brighterscript instead of hardcoding them in the AST itself. However, thinking more about this, I don't think this specific implementation should be in the plugin. So, ignore the `BrsFilePreTranspileProcessor` comment. This should be moved into the `NewExpression` transpile function.